src/main/java/xyz/nkomarn/harbor/listener/AfkListener.java

```diff
@@ -0,0 +1,150 @@
+package xyz.nkomarn.harbor.listener;
+
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.Listener;
+import org.bukkit.event.inventory.InventoryClickEvent;
+import org.bukkit.event.player.AsyncPlayerChatEvent;
+import org.bukkit.event.player.PlayerCommandPreprocessEvent;
+import org.bukkit.event.player.PlayerJoinEvent;
+import org.bukkit.event.player.PlayerQuitEvent;
+import org.bukkit.plugin.java.JavaPlugin;
+import org.bukkit.scheduler.BukkitRunnable;
+import xyz.nkomarn.harbor.Harbor;
+import xyz.nkomarn.harbor.provider.DefaultAFKProvider;
+
+import java.util.ArrayDeque;
+import java.util.Queue;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public final class AfkListener implements Listener {
+    private final DefaultAFKProvider afkProvider;
+    private Queue<AfkPlayer> players;
+    private PlayerMovementChecker movementChecker;
+
+    public AfkListener(DefaultAFKProvider afkProvider) {
+        this.afkProvider = afkProvider;
+        JavaPlugin.getPlugin(Harbor.class).getLogger().info("Initializing fallback AFK detection system. Fallback AFK system is not enabled at this time");
+    }
+
+    /**
+     * Provides a way to start the listener
+     */
+    public void start() {
+        JavaPlugin plugin = JavaPlugin.getPlugin(Harbor.class);
```

Comment:
Please pass the Harbor instance trough the constructor, to maintain consistency with the rest of the code.